app/grandchallenge/cases/tasks.py

```diff
@@ -229,9 +230,24 @@ def fix_mhd_file(file, prefix):
         pass
 
 
-def extract_and_flatten(file, path, index, prefix="", is_tar=False):
+def _check_sanity(info, is_tar, path):
+    # Check tar files for symlinks and reject upload session if any.
+    if is_tar:
+        if info.issym() or info.islnk():
+            raise ValidationError("(Symbolic) links are not allowed.")
+    # Also check for max path length, drop folders when path is too long
+    filename_attr = "name" if is_tar else "filename"
+    while len(os.path.join(path, getattr(info, filename_attr))) > 4095:
+        filename = getattr(info, filename_attr)
+        filename = os.path.join(
+            *Path(filename).parts[1 : len(Path(filename).parts)]
+        )
+        setattr(info, filename_attr, filename)
```

Comment:
Style wise: split these out into 2 more functions rather than using comments. Fine for now.